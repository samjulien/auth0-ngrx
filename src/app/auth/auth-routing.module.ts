import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { LoginPageComponent } from './components/login-page.component';
import { UserHomeComponent } from './components/user-home.component';
import { AuthGuardService } from './services/auth-guard.service';
import { CallbackComponent } from '@app/auth/components/callback.component';

@NgModule({
  imports: [
    RouterModule.forChild([
      { path: 'login', component: LoginPageComponent },
      { path: 'callback', component: CallbackComponent },
      {
        path: 'home',
        component: UserHomeComponent,
        canActivate: [AuthGuardService]
      }
    ])
  ]
})
export class AuthRoutingModule {}
