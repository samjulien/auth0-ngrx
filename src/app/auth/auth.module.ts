import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import {
  MatInputModule,
  MatButtonModule,
  MatCardModule,
  MatDialogModule
} from '@angular/material';
import { EffectsModule } from '@ngrx/effects';
import { AuthService } from './services/auth.service';
import { AuthEffects } from './effects/auth.effects';
import { LoginPageComponent } from './components/login-page.component';
import { AuthRoutingModule } from './auth-routing.module';
import { LoginFormComponent } from './components/login-form.component';
import { LogoutPromptComponent } from './components/logout-prompt.component';
import { UserHomeComponent } from '@app/auth/components/user-home.component';
import { CallbackComponent } from '@app/auth/components/callback.component';

@NgModule({
  imports: [
    CommonModule,
    AuthRoutingModule,
    MatButtonModule,
    MatCardModule,
    MatDialogModule,
    MatInputModule,
    ReactiveFormsModule,
    EffectsModule.forFeature([AuthEffects])
  ],
  declarations: [
    LoginPageComponent,
    LoginFormComponent,
    LogoutPromptComponent,
    UserHomeComponent,
    CallbackComponent
  ],
  entryComponents: [LogoutPromptComponent]
})
export class AuthModule {}
