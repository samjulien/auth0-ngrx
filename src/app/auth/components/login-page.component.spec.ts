import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import {
  createComponentFixture,
  provideMagicalMock
} from 'angular-testing-library';
import { LoginPageComponent } from './login-page.component';
import { LoginFormComponent } from './login-form.component';
import { Store } from '@ngrx/store';
import { BehaviorSubject } from 'rxjs';
import { Login } from '../actions/auth.actions';

describe('Login Page Component', () => {
  const fixture = createComponentFixture({
    component: LoginPageComponent,
    declarations: [LoginFormComponent],
    providers: [provideMagicalMock(Store)]
  });

  const error$ = new BehaviorSubject<string>('');
  const pending$ = new BehaviorSubject<boolean>(false);
  let store: Store<any>;

  beforeEach(async () => {
    await fixture.compile({
      error$,
      pending$
    });

    store = fixture.get(Store);
  });

  it('should compile', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should dispatch an action when onLogin is called', () => {
    const credentials = {
      username: 'test',
      password: 'test'
    };

    fixture.instance.onLogin();

    expect(store.dispatch).toHaveBeenCalledWith(new Login());
  });
});
