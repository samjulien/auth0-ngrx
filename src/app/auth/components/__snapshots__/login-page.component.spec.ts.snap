// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page Component should compile 1`] = `
<app-login-page
  error$={[Function BehaviorSubject]}
  pending$={[Function BehaviorSubject]}
  store={[Function Object]}
>
  <app-login-form
    _ngcontent-c0=""
    _nghost-c1=""
    ng-reflect-disabled="false"
    ng-reflect-error=""
  >
    <mat-card
      _ngcontent-c1=""
    >
      <mat-card-title
        _ngcontent-c1=""
      >
         Login 
      </mat-card-title>
      <mat-card-content
        _ngcontent-c1=""
      >
        <form
          _ngcontent-c1=""
        >
          <mat-form-field
            _ngcontent-c1=""
          >
            <input
              _ngcontent-c1=""
              formcontrolname="username"
              matinput=""
              placeholder="Username"
              type="text"
            />
            
            <mat-error
              _ngcontent-c1=""
            >
              Username is required
            </mat-error>
          </mat-form-field>
          <mat-form-field
            _ngcontent-c1=""
          >
            <input
              _ngcontent-c1=""
              formcontrolname="password"
              matinput=""
              placeholder="Password"
              type="password"
            />
            
            <mat-error
              _ngcontent-c1=""
            >
              Password is required
            </mat-error>
          </mat-form-field>
          
          <div
            _ngcontent-c1=""
            class="loginButtons"
          >
            <button
              _ngcontent-c1=""
              mat-button=""
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </mat-card-content>
    </mat-card>
  </app-login-form>
</app-login-page>
`;
